java-manta-client/src/test/java/com/joyent/manta/client/crypto/AesCtrCipherDetailsTest.java

```diff
@@ -176,15 +176,15 @@ protected void canRandomlyReadPlaintextPositionFromCiphertext(final SecretKey se
                 decryptor.init(Cipher.DECRYPT_MODE, secretKey, cipherDetails.getEncryptionParameterSpec(iv));
                 long adjustedPlaintextRange = cipherDetails.updateCipherToPosition(decryptor, startPlaintextRange);
 
-                byte[] adjustedCipherText = Arrays.copyOfRange(ciphertext, (int) startCipherTextRange, (int) endCipherTextRange);
+                byte[] adjustedCipherText = Arrays.copyOfRange(ciphertext, (int) startCipherTextRange, (int) Math.min(ciphertext.length, endCipherTextRange + 1));
                 byte[] out = decryptor.doFinal(adjustedCipherText);
                 byte[] decrypted = Arrays.copyOfRange(out, (int) adjustedPlaintextRange,
-                        (int) adjustedPlaintextLength + (int) adjustedPlaintextRange);
+                        (int) Math.min(out.length, adjustedPlaintextLength));
```

Comment:
Exactly, since the bytes for all blocks are requested, and AES-CTR isn't padded, more bytes than exist may be requested using the above calculations.  There's no way to know this without knowing content length (HTTP request).

The content length (HTTP request) isn't necessary for this though.  Per Range Request RFC (7233) it is *not* an error to request more bytes than exist, and only the existing bytes are returned (which is emulated using `Math.min`).